File: gto-support-jsonapi-retrofit2/src/main/java/org/ccci/gto/android/common/jsonapi/retrofit2/model/JsonApiRetrofitObject.java
package org.ccci.gto.android.common.jsonapi.retrofit2.model;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;

import org.ccci.gto.android.common.jsonapi.JsonApiConverter.Options;
import org.ccci.gto.android.common.jsonapi.model.JsonApiObject;

public class JsonApiRetrofitObject<T> extends JsonApiObject<T> {
    @Nullable
    private Options mOptions;

    public JsonApiRetrofitObject(@NonNull final JsonApiObject<T> obj) {
        super(obj);
    }

    @NonNull
    public static <T> JsonApiRetrofitObject<T> single(@Nullable final T data) {
        return new JsonApiRetrofitObject<>(JsonApiObject.single(data));
    }

    @Nullable
    public Options getOptions() {
        return mOptions;
    }

    public void setOptions(@Nullable final Options options) {
        mOptions = options;
    }
}

File: gto-support-jsonapi-retrofit2/src/main/kotlin/org/ccci/gto/android/common/jsonapi/retrofit2/model/JsonApiRetrofitObject.kt
package org.ccci.gto.android.common.jsonapi.retrofit2.model

import org.ccci.gto.android.common.jsonapi.JsonApiConverter
import org.ccci.gto.android.common.jsonapi.model.JsonApiObject

class JsonApiRetrofitObject<T : Any>(obj: JsonApiObject<T>) : JsonApiObject<T>(obj) {
    companion object {
        fun <T : Any> single(data: T?) = JsonApiRetrofitObject(JsonApiObject.single(data))
    }

    var options: JsonApiConverter.Options? = null
}
